src/sentry/static/sentry/app/views/performance/transactionDetails/quickTraceQuery.tsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import omit from 'lodash/omit';
+
+import {Client} from 'app/api';
+import {getTraceDateTimeRange} from 'app/components/events/interfaces/spans/utils';
+import {ALL_ACCESS_PROJECTS} from 'app/constants/globalSelectionHeader';
+import {Event, EventTransaction} from 'app/types/event';
+import EventView from 'app/utils/discover/eventView';
+import GenericDiscoverQuery, {
+  DiscoverQueryProps,
+  GenericChildrenProps,
+} from 'app/utils/discover/genericDiscoverQuery';
+import withApi from 'app/utils/withApi';
+
+import {isTransaction} from './utils';
+
+type TransactionLite = {
+  event_id: string;
+  span_id: string;
+  transaction: string;
+  project_id: number;
+  parent_event_id: string | null;
+  is_root: boolean;
+};
+
+export type TraceLite = TransactionLite[];
+
+type QuickTraceProps = {
+  event: Event;
+};
+
+type RequestProps = DiscoverQueryProps & QuickTraceProps;
+
+type ChildrenProps = Omit<GenericChildrenProps<QuickTraceProps>, 'tableData'> & {
+  trace: TraceLite | null;
+};
+
+type QueryProps = Omit<RequestProps, 'eventView'> & {
+  children: (props: ChildrenProps) => React.ReactNode;
+};
+
+function getQuickTraceRequestPayload({eventView, event, location}: RequestProps) {
+  const additionalApiPayload = omit(eventView.getEventsAPIPayload(location), [
+    'field',
+    'sort',
+    'per_page',
+  ]);
+  return Object.assign({event_id: event.id}, additionalApiPayload);
+}
+
+function beforeFetch(api: Client) {
+  api.clear();
+}
+
+function makeEventView(event: EventTransaction) {
+  // TODO: increase the time range here for a buffer around the start/end of
+  // the current transactions to find other transactions in the trace better
```

Comment:
Is this TODO still accurate? Since `getTraceDateTimeRange` adds a 12h buffer

Reply:
Oops, thought I removed it. Should be gone in 45c5ed3